stl/inc/xutility

```diff
@@ -5148,8 +5132,29 @@ _INLINE_VAR constexpr bool _Memchr_in_find_is_safe =
             > && !is_volatile_v<remove_reference_t<_Iter_ref_t<_Iter>>>;
 
 template <class _InIt, class _Ty>
-_NODISCARD constexpr _InIt _Find_unchecked1(_InIt _First, const _InIt _Last, const _Ty& _Val, false_type) {
-    // find first matching _Val
+_NODISCARD _CONSTEXPR20 _InIt _Find_unchecked(_InIt _First, const _InIt _Last, const _Ty& _Val) {
+    // find first matching _Val; choose optimization
+    // activate optimization for contiguous iterators to (const) bytes and integral values
+    // find first byte matching integral _Val
+#if _HAS_CXX20
+    if (!_STD is_constant_evaluated())
+#endif // _HAS_CXX20
+    {
+        if constexpr (_Memchr_in_find_is_safe<_InIt, _Ty>) {
+            if (!_Within_limits(_First, _Val)) {
+                return _Last;
+            }

```

Comment:
No change requested: I observe that originally, in C++20 when `is_constant_evaluated()` was `true`, we would perform the `_Within_limits` check and early return if possible. Now, when `is_constant_evaluated()` is `true`, we simply activate the classic algorithm fallback. This seems fine, since at most there would be a minor throughput improvement, and it seems fairly unusual for the `_Within_limits` check to actually be successful (when it isn't, we pay the cost of a branch).

That is, `_Within_limits` is necessary for the runtime `memchr` optimization, but it's not really valuable for the `constexpr` case, and the new code is organized accordingly. :smile_cat: